plugins/net_plugin/net_plugin.cpp

```diff
@@ -2228,13 +2246,18 @@ namespace eosio {
 
    template<typename VerifierFunc>
    void net_plugin_impl::send_transaction_to_all(const std::shared_ptr<std::vector<char>>& send_buffer, VerifierFunc verify) {
-      for( auto &c : connections) {
-         if( c->current() && verify( c ) && !c->is_gray && !c->considers_gray) {
-            c->enqueue_buffer( send_buffer, true, priority::low, no_reason );
+      for (auto itr = connections.lower_bound(connection_state::syncing); itr != connections.end(); ++itr) {
+         if (verify(*itr) && !(*itr)->is_gray && !(*itr)->considers_gray) {
+            (*itr)->enqueue_buffer(send_buffer, true, priority::low, no_reason);
          }
       }
    }
 
+   void net_plugin_impl::emplace_connection(connection_ptr& conn) {
+      auto itr = connections.emplace(conn).first;
+      conn->bmi_modify = [&](auto&& func) { connections.modify(itr, func); };
```

Comment:
Looks good! I really like it.
But the original sources already has decision: there is global variable to the `net_plugin_impl` -> my_impl:
https://github.com/cyberway/cyberway/blob/48c45b10a212a679d08a7a7ca29e2e2785432f39/plugins/net_plugin/net_plugin.cpp#L1352

Reply:
@afalaleev okay, now I used it, but `itr` still should be set to each connection. Or you want I finding it in multi_index each time when I need it? It is not effective for performance.